src/rpc/server.cpp

```diff
@@ -444,7 +444,7 @@ static inline JSONRPCRequest transformNamedArguments(const JSONRPCRequest& in, c
     const std::vector<UniValue>& values = in.params.getValues();
     std::unordered_map<std::string, const UniValue*> argsIn;
     for (size_t i=0; i<keys.size(); ++i) {
-        argsIn[keys[i]] = &values[i];
+        argsIn[keys[i]] = values.data() + i;
```

Comment:
Is this really clearer?

Reply:
Agree with @jtimon here. This PR is titled `Changing &var[0] to var.data()`, this seems to overshoot that scope. Addressing out of an vector is UB no matter what, so this doesn't fix any potential issues, and isn't more readable IMO either.